Lesson17/gokce/contactBook.js

```diff
@@ -0,0 +1,279 @@
+    if (newName === contactList[i].name) {
+      console.warn("The contact already exist");
+      return;
+    }
+  }
+  const contactListLengthPrev = contactList.length;
+  contactList.push({
+    name: newName,
+    phone: newPhone,
+    email: newEmail,
+  });
+  const contactListLengthNew = contactList.length;
+
+  if (contactListLengthPrev < contactListLengthNew) {
+    console.log("Contact added successfully.");
+  } else {
+    console.log("An error occured when adding a contact.");
+  }
+
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator
+  // contactListLengthPrev < contactListLengthNew
+  //   ? console.log("Contact added successfully.")
+  //   : console.log("An error occured when adding a contact.");
+}
+
+/*
+-----------------------------------------------------------
+  STEP 4: View a Contact by Name
+-----------------------------------------------------------
+Function: viewContact(name)
+- Loops over 'contacts' to find one matching 'name'.
+- Logs the contact info if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+
+
+
+function viewContact(name, contactList) {
+  let isContactFound = false;
+  for (const contact of contactList) {
+    if (name === contact.name) {
+      console.log(
+        `name: ${contact.name}, phone:${contact.phone}, email:${contact.email}`,
+      );
+      isContactFound = true;
+    }
+  }
+  if (!isContactFound) {
+    console.log(`No contact found with the name: ${name}`);
+  }
+}
+
+/*
+-----------------------------------------------------------
+  STEP 5: Update a Contact
+-----------------------------------------------------------
+Function: updateContact(name, newPhone, newEmail)
+- Finds the contact by name and updates phone + email.
+- Logs "Contact updated successfully." if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+/*
+function updateContact(name,newPhone,newEmail,contactList){
+  let isFoundName=false
+  for(const contact of contactList){
+    if(name===contact.name){
+      contact.phone=newPhone;
+      contact.email=newEmail;
+      isFoundName=true;
+      console.log("Contact updated successfully.");
+    }
+  }
+  if(!isFoundName){
+    console.log(`No contact found with the name:${name}`);
+  }
+}
+
+//second option
+function findContact(name, contactList) {
+  for (const contact of contactList) {
+    if (contact.name === name) {
+      return contact; 
+    }
+  }
+  return null; 
+}
+
+function updateContact(name, newPhone, newEmail, contactList) {
+  const contact = findContact(name, contactList); 
+  if (contact) { 
+    contact.phone = newPhone; 
+    contact.email = newEmail;
+    console.log(`Contact updated successfully: ${contact.name}, ${contact.phone}, ${contact.email}`);
+  } else { // Eğer kişi bulunmazsa
+    console.log(`No contact found with the name: ${name}`);
+  }
+}
+*/
+//third option
+
+function findContact(name, contactList) {
```

Comment:
Move this function to the top of the file (above other functions) otherwise you'll get a warning that findContact function is used before its declaration (when used in the addContact and viewContact).